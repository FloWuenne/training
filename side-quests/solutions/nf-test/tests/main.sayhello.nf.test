nextflow_process {

    name "Test Process sayHello"
    script "main.nf"
    process "sayHello"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = "hello"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
