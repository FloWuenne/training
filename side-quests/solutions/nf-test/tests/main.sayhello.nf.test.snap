{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hello-output.txt:md5,b1946ac92492d2347c6235b4d2611184"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-19T11:52:09.690504479"
    }
}